@azure/service-bus doesn't tsc transpile

When I try to use `tsc` while importing the `@azure/service-bus` packages I get the following errors:

> node_modules/@azure/service-bus/typings/lib/subscriptionClient.d.ts:1:23 - error TS2688: Cannot find type definition file for 'long'.
>
> 1 // \<reference types="long" />
>                         ~~~~
> 
> node_modules/@azure/service-bus/typings/lib/subscriptionClient.d.ts:76:46 - error TS2304: Cannot find name 'Long'.
> 
> 76     peekBySequenceNumber(fromSequenceNumber: Long, messageCount?: number): Promise<ReceivedMessageInfo[]>;
>                                                 ~~~~
> 
> 
> Found 2 errors.

Ok installed both `@types/node` and `@types/long` and get this error:


> node_modules/@azure/service-bus/typings/lib/core/managementClient.d.ts:1:8 - error TS1192: Module '"/Users/jeffhollan/Development/GitHub/functions-multiple-queue-exponential/functions-node-queue-exponential/node_modules/@types/long/index"' has no default export.
> 
> 1 import Long from "long";
>          ~~~~
> 
> node_modules/@azure/service-bus/typings/lib/serviceBusMessage.d.ts:2:8 - error TS1192: Module '"/Users/jeffhollan/Development/GitHub/functions-multiple-queue-exponential/functions-node-queue-exponential/node_modules/@types/long/index"' has no default export.
> 
> 2 import Long from "long";
>          ~~~~
> 
> 
> Found 2 errors.

--- a/packages/@azure/servicebus/data-plane/README.md
+++ b/packages/@azure/servicebus/data-plane/README.md
@@ -1,48 +1,65 @@
-@azure/service-bus
-================
+# @azure/service-bus
 
 This library provides a convenient way to interact with [Azure Service Bus](https://azure.microsoft.com/en-us/services/service-bus/).
 
-## Status ##
+## Status
 
 This library is currently in preview and may change prior to release.
 
-## Pre-requisite ##
-- **Node.js version: 6.x or higher.** 
+## Pre-requisite
 
-## Installation ##
+- **Node.js version: 6.x or higher.**
+
+## Installation
+
 ```bash
 npm install @azure/service-bus
 ```
 
-## Examples 
+TypeScript users need to install Node types:
 
-Please take a look at the [examples](https://github.com/Azure/azure-sdk-for-js/tree/master/packages/%40azure/servicebus/data-plane/examples) 
-directory for detailed examples on how to use this library to send and receive messages to/from 
+```bash
+npm install @types/node
+```
+
+And also enable `compilerOptions.esModuleInterop` in tsconfig.json.
+
+## Examples
+
+Please take a look at the [examples](https://github.com/Azure/azure-sdk-for-js/tree/master/packages/%40azure/servicebus/data-plane/examples)
+directory for detailed examples on how to use this library to send and receive messages to/from
 [Service Bus Queues, Topics and Subscriptions](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-messaging-overview).
 
-## Debug logs ##
+## Debug logs
 
 You can set the following environment variable to get the debug logs when using this library.
 
 - Getting debug logs from the Service Bus SDK
+
 ```bash
 export DEBUG=azure*
 ```
+
 - Getting debug logs from the Service Bus SDK and the protocol level library.
+
 ```bash
 export DEBUG=azure*,rhea*
 ```
+
 - If you are **not interested in viewing the message transformation** (which consumes lot of console/disk space) then you can set the `DEBUG` environment variable as follows:
+
 ```bash
 export DEBUG=azure*,rhea*,-rhea:raw,-rhea:message,-azure:amqp-common:datatransformer
 ```
+
 - If you are interested only in **errors**, then you can set the `DEBUG` environment variable as follows:
+
 ```bash
 export DEBUG=azure:service-bus:error,azure-amqp-common:error,rhea-promise:error,rhea:events,rhea:frames,rhea:io,rhea:flow
 ```
 
 #### Logging to a file
+
 - Set the `DEBUG` environment variable as shown above and then run your test script as follows:
   - Logging statements from you test script go to `out.log` and logging statement from the sdk go to `debug.log`.
     ```bash
@@ -56,4 +73,3 @@ export DEBUG=azure:service-bus:error,azure-amqp-common:error,rhea-promise:error,
     ```bash
       node your-test-script.js &> out.log
     ```
-

--- a/packages/@azure/servicebus/data-plane/README.md
+++ b/packages/@azure/servicebus/data-plane/README.md
@@ -16,13 +16,15 @@ This library is currently in preview and may change prior to release.
 npm install @azure/service-bus
 ```
 
-TypeScript users need to install Node types:
+### TypeScript
+
+TypeScript users need to have Node type definitions installed:
 
 ```bash
 npm install @types/node
 ```
 
-And also enable `compilerOptions.esModuleInterop` in tsconfig.json.
+You also need to enable `compilerOptions.allowSyntheticDefaultImports` in your tsconfig.json. Note that if you have enabled `compilerOptions.esModuleInterop`, `allowSyntheticDefaultImports` is enabled by default.
 
 ## Examples
 

--- a/packages/@azure/servicebus/data-plane/README.md
+++ b/packages/@azure/servicebus/data-plane/README.md
@@ -24,7 +24,7 @@ TypeScript users need to have Node type definitions installed:
 npm install @types/node
 ```
 
-You also need to enable `compilerOptions.allowSyntheticDefaultImports` in your tsconfig.json. Note that if you have enabled `compilerOptions.esModuleInterop`, `allowSyntheticDefaultImports` is enabled by default.
+You also need to enable `compilerOptions.allowSyntheticDefaultImports` in your tsconfig.json. Note that if you have enabled `compilerOptions.esModuleInterop`, `allowSyntheticDefaultImports` is enabled by default. See [TypeScript's compiler options handbook](https://www.typescriptlang.org/docs/handbook/compiler-options.html) for more information.
 
 ## Examples
 

--- a/packages/@azure/servicebus/data-plane/lib/index.ts
+++ b/packages/@azure/servicebus/data-plane/lib/index.ts
@@ -1,6 +1,8 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+/// <reference lib="es2015" />
+
 export { Namespace, NamespaceOptions, NamespaceOptionsBase } from "./namespace";
 export {
   TokenInfo,

--- a/packages/@azure/servicebus/data-plane/lib/util/utils.ts
+++ b/packages/@azure/servicebus/data-plane/lib/util/utils.ts
@@ -7,6 +7,13 @@ import { generate_uuid, string_to_uuid } from "rhea-promise";
 import { isBuffer } from "util";
 import { ConnectionContext } from "../connectionContext";
 
+// This is the only dependency we have on DOM types, so rather than require
+// the DOM lib we can just shim this in.
+interface Navigator {
+  hardwareConcurrency: number;
+}
+declare var navigator: Navigator;
+
 /**
  * A constant that indicates whether the environment is node.js or browser based.
  */

--- a/packages/@azure/servicebus/data-plane/package.json
+++ b/packages/@azure/servicebus/data-plane/package.json
@@ -28,7 +28,9 @@
     "long": "^4.0.0",
     "ms-rest-azure": "^2.5.9",
     "rhea-promise": "^0.1.13",
-    "tslib": "^1.9.3"
+    "tslib": "^1.9.3",
+    "@types/node": "^8.0.37",
+    "@types/long": "^4.0.0"
   },
   "devDependencies": {
     "@azure/arm-servicebus": "^0.1.0",
@@ -39,9 +41,7 @@
     "@types/chai-as-promised": "^7.1.0",
     "@types/debug": "^0.0.31",
     "@types/dotenv": "^4.0.3",
-    "@types/long": "^4.0.0",
     "@types/mocha": "^5.2.5",
-    "@types/node": "^8.0.37",
     "chai": "^4.2.0",
     "chai-as-promised": "^7.1.1",
     "cross-env": "^5.2.0",

--- a/packages/@azure/servicebus/data-plane/package.json
+++ b/packages/@azure/servicebus/data-plane/package.json
@@ -23,14 +23,13 @@
   },
   "dependencies": {
     "@azure/amqp-common": "^1.0.0-preview.1",
+    "@types/long": "^4.0.0",
     "debug": "^3.1.0",
     "is-buffer": "^2.0.3",
     "long": "^4.0.0",
     "ms-rest-azure": "^2.5.9",
     "rhea-promise": "^0.1.13",
-    "tslib": "^1.9.3",
-    "@types/node": "^8.0.37",
-    "@types/long": "^4.0.0"
+    "tslib": "^1.9.3"
   },
   "devDependencies": {
     "@azure/arm-servicebus": "^0.1.0",
@@ -42,6 +41,7 @@
     "@types/debug": "^0.0.31",
     "@types/dotenv": "^4.0.3",
     "@types/mocha": "^5.2.5",
+    "@types/node": "^11.9.5",
     "chai": "^4.2.0",
     "chai-as-promised": "^7.1.1",
     "cross-env": "^5.2.0",

--- a/packages/@azure/servicebus/data-plane/package.json
+++ b/packages/@azure/servicebus/data-plane/package.json
@@ -41,7 +41,7 @@
     "@types/debug": "^0.0.31",
     "@types/dotenv": "^4.0.3",
     "@types/mocha": "^5.2.5",
-    "@types/node": "^11.9.5",
+    "@types/node": "^8.0.37",
     "chai": "^4.2.0",
     "chai-as-promised": "^7.1.1",
     "cross-env": "^5.2.0",

--- a/packages/@azure/servicebus/data-plane/tsconfig.json
+++ b/packages/@azure/servicebus/data-plane/tsconfig.json
@@ -3,24 +3,22 @@
     /* Basic Options */
     "target": "es6" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */,
     "module": "es6" /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */,
-
+    "lib": [] /* lib dependencies are triple-slash directives in lib/index.ts */,
     "declaration": true /* Generates corresponding '.d.ts' file. */,
     "declarationMap": true /* Generates a sourcemap for each corresponding '.d.ts' file. */,
     "sourceMap": true /* Generates corresponding '.map' file. */,
-    
-    "outDir": "./dist-esm" /* Redirect output structure to the directory. */,
-    "declarationDir": "./typings", /* Output directory for generated declaration files.*/
- 
-    "stripInternal": true, /* Do not emit declarations for code with @internal annotation*/
 
+    "outDir": "./dist-esm" /* Redirect output structure to the directory. */,
+    "stripInternal": true /* Do not emit declarations for code with @internal annotation*/,
+    "declarationDir": "./typings" /* Output directory for generated declaration files.*/,
     "importHelpers": true /* Import emit helpers from 'tslib'. */,
-    
+
     /* Strict Type-Checking Options */
     "strict": true /* Enable all strict type-checking options. */,
-    "noImplicitReturns": true, /* Report error when not all code paths in function return a value. */
+    "noImplicitReturns": true /* Report error when not all code paths in function return a value. */,
 
     /* Additional Checks */
-    "noUnusedLocals": true,                /* Report errors on unused locals. */
+    "noUnusedLocals": true /* Report errors on unused locals. */,
 
     /* Module Resolution Options */
     "moduleResolution": "node" /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */,
@@ -31,17 +29,10 @@
     "forceConsistentCasingInFileNames": true,
 
     /* Other options */
-    "newLine": "LF", /*	Use the specified end of line sequence to be used when emitting files: "crlf" (windows) or "lf" (unix).”*/
-    "allowJs": false, /* Don't allow JavaScript files to be compiled.*/
+    "newLine": "LF" /*	Use the specified end of line sequence to be used when emitting files: "crlf" (windows) or "lf" (unix).”*/,
+    "allowJs": false /* Don't allow JavaScript files to be compiled.*/
   },
   "compileOnSave": true,
-  "exclude": [
-    "node_modules",
-    "typings/**",
-    "./examples/**/*.ts"
-  ],
-  "include": [
-    "./lib/**/*.ts",
-    "./test/**/*.ts",
-  ]
+  "exclude": ["node_modules", "typings/**", "./examples/**/*.ts"],
+  "include": ["./lib/**/*.ts", "./test/**/*.ts"]
 }

